Userland/Libraries/LibWeb/FileAPI/FileReader.cpp

```diff
@@ -0,0 +1,359 @@
+/*
+ * Copyright (c) 2023, Shannon Booth <shannon@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/Assertions.h>
+#include <AK/Base64.h>
+#include <AK/ByteBuffer.h>
+#include <LibJS/Heap/Heap.h>
+#include <LibJS/Runtime/Promise.h>
+#include <LibJS/Runtime/Realm.h>
+#include <LibJS/Runtime/TypedArray.h>
+#include <LibTextCodec/Decoder.h>
+#include <LibWeb/Bindings/Intrinsics.h>
+#include <LibWeb/DOM/Event.h>
+#include <LibWeb/DOM/EventTarget.h>
+#include <LibWeb/FileAPI/Blob.h>
+#include <LibWeb/FileAPI/FileReader.h>
+#include <LibWeb/HTML/EventLoop/EventLoop.h>
+#include <LibWeb/HTML/EventNames.h>
+#include <LibWeb/HTML/Scripting/TemporaryExecutionContext.h>
+#include <LibWeb/MimeSniff/MimeType.h>
+#include <LibWeb/Platform/EventLoopPlugin.h>
+#include <LibWeb/Streams/AbstractOperations.h>
+#include <LibWeb/Streams/ReadableStream.h>
+#include <LibWeb/Streams/ReadableStreamDefaultReader.h>
+#include <LibWeb/WebIDL/DOMException.h>
+#include <LibWeb/WebIDL/ExceptionOr.h>
+
+namespace Web::FileAPI {
+
+FileReader::~FileReader() = default;
+
+FileReader::FileReader(JS::Realm& realm)
+    : DOM::EventTarget(realm)
+{
+}
+
+void FileReader::initialize(JS::Realm& realm)
+{
+    DOM::EventTarget::initialize(realm);
```

Comment:
Normally we spell this as "Base" in initialize. The using def is added in the PLATFORM_OBJECT macro. (or the JS_CELL macro, idk which)

Reply:
Oh nice - thanks!